### DISCUSSION TITLE
Reown wallet issue in telegram mini app (Chain switch )

### DISCUSSION THREAD

**Discussion by shubham-singh-antie** [WalletConnect/walletconnect-monorepo#5573] [Developer Support]
I am integrating the Reown wallet into my Telegram mini-app and facing two major issues:

Unable to add or switch chains in MetaMask
Case: If the user does not have the network added in MetaMask, the switch network function does not allow me to switch the chain.

Unable to detect the active chain in MetaMask
Case: If the user already has the network added in MetaMask but has selected a different network, I cannot determine which chain is currently active. When calling any function, it throws an error incorrect chain Id.

If user have added network manually and network is selected. in this case everything is working fine.

**Comment by rtomas** [comment]
hi @shubham-singh-antie .. I think we already talked over Discord. The issue has been reported.
